src/routes/+page.svelte

```diff
@@ -8,31 +8,76 @@
 
 	let arrivalsAndDepartures = $state([]);
 	let loading = $state(true);
+	let currentDateTime = $state(new Date());
 
-	// Calculate arrival time in minutes
+	// Format the current date and time
+	function formatCurrentDateTime(date) {
+		const options = {
+			weekday: 'long',
+			month: 'long',
+			day: 'numeric'
+		};
+		return date.toLocaleDateString('en-US', options);
+	}
+
+	// Format the current time
+	function formatCurrentTime(date) {
+		return date.toLocaleTimeString('en-US', {
+			hour: '2-digit',
+			minute: '2-digit',
+			second: '2-digit',
+			hour12: true
+		});
+	}
+
+	// Get status for arrival or departure
 	function getArrivalStatus(predictedTime, scheduledTime) {
```

Comment:
please extract this into a separate file (probably formatters.js for now?) and add a unit test or two to it.